apps/emqx_authn/src/simple_authn/emqx_authn_pgsql.erl

```diff
@@ -115,7 +115,7 @@ authenticate(#{password := Password} = Credential,
                placeholders := PlaceHolders,
                resource_id := ResourceId} = State) ->
     Params = emqx_authn_utils:replace_placeholders(PlaceHolders, Credential),
-    case emqx_resource:query(ResourceId, {sql, Query, Params}) of
+    case emqx_resource:query(ResourceId, {prepared_sql, Query, Params}) of
```

Comment:
Shouldn't there be an option to run queries without prepare, for the cases when clients have e.g. pgbouncer with transaction pooling?

Then `epgsql:parse` and subsequent `epgsql:prepared_query` can go to different sessions.

Reply:
If you want to run queries without prepare, you can call like this: `emqx_resource:query(ResourceId, {sql, Query, Params})`.

This should be what you are talking about? I'm not sure if I understand what you mean correctly

Reply:
Sorry, 
I mean a config option for authn/authz configuration.

Reply:
This option should not be exposed to users. Using `prepared_query` is a prerequisite for supporting high concurrency, otherwise `epgsql` will report an error when high concurrency occurs. Related issue: https://github.com/emqx/emqx-auth-pgsql/issues/84

Reply:
Seems that users experience the described problem, when they have postgresql behind pgbouncer with transaction pooling https://github.com/emqx/emqx-auth-pgsql/issues/197 🤔 

Since there is a separate level of pooling behind pgbouncer, we cannot be sure that `prepare` and subsequent query come to the same postgresql session, even if they are sent to the same ecpool worker.

Reply:
I don't know much about pgbouncer, but as far as this issue is concerned, it should be because `epgsql:parse` was not called again after reconnecting, we have solved it in subsequent versions.